File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/parser/ParsedMovement.java
package br.com.calciolari.datahub.imports.domain.parser;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Objects;

/**
 * Normalized source movement. Money and quantities are {@link BigDecimal} only.
 */
public record ParsedMovement(
		int sourceRecordIndex,
		MovementDirection direction,
		String externalProductId,
		String productName,
		String externalSaleId,
		LocalDateTime occurredAt,
		BigDecimal quantity,
		BigDecimal unitPrice,
		BigDecimal discountPercentage,
		BigDecimal total,
		BigDecimal previousStock,
		BigDecimal resultingStock,
		String manufacturer,
		SourceLocator sourceLocator
) {
	public ParsedMovement {
		Objects.requireNonNull(direction, "direction");
		if (sourceLocator == null) {
			sourceLocator = SourceLocator.empty();
		}
	}
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Parser/ParsedMovement.cs
namespace Calciolari.DataHub.Imports.Domain.Parser;

/// <summary>
/// Normalized source movement. Money and quantities are decimal only.
/// </summary>
public sealed record ParsedMovement(
    int SourceRecordIndex,
    MovementDirection Direction,
    string? ExternalProductId,
    string? ProductName,
    string? ExternalSaleId,
    DateTime? OccurredAt,
    decimal? Quantity,
    decimal? UnitPrice,
    decimal? DiscountPercentage,
    decimal? Total,
    decimal? PreviousStock,
    decimal? ResultingStock,
    string? Manufacturer,
    SourceLocator SourceLocator);
